Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -360,15 +368,26 @@ DetectorInfo::buildMergeIndices(const DetectorInfo &other) const {
   // TODO If we make masking time-independent we need to check masking here.
 
   if (m_isSyncScan) {
-    for (const auto &interval1 : *other.m_scanIntervals) {
-      for (const auto &interval2 : *m_scanIntervals) {
-        if (!((interval1.second <= interval2.first) ||
-              (interval1.first >= interval2.second))) {
-          failMerge("scan intervals overlap in sync scan");
+    std::vector<bool> merge(other.m_scanIntervals->size(), true);
+
+    for (size_t t1 = 0; t1 < other.m_scanIntervals->size(); ++t1) {
+      for (size_t t2 = 0; t2 < m_scanIntervals->size(); ++t2) {
+        if ((*other.m_scanIntervals)[t1] == (*m_scanIntervals)[t2]) {
+          for (size_t detIndex = 0; detIndex < size(); ++detIndex) {
```

Comment:
Doing the check pixel-by-pixel is probably slower than it would be by comparing ranges in vectors of data, but I suppose you do not care since identical intervals are an edge case that is not common in usual operations?

Reply:
Yes, exactly. In the majority of cases you would have different, non-overlapping scan intervals. In the cases where you do have identical intervals I think the behaviour for sync and async should be the same.